messaging/controller/processor/processor.go

```
@@ -282,6 +287,22 @@ func (r *Processor) processEncryptionLayer(m *types.Message, logger *zap.Logger)
 	return err
 }
 
+func (r *Processor) processSDSLayer(msg *types.Message) error {
+	if len(msg.EncryptionLayer.Payload) > 0 {
+		unwrappedMessage, err := r.stack.SDSManager.UnwrapReceivedMessage(msg.EncryptionLayer.Payload)
+		if err != nil {
+			r.logger.Warn("SDS: failed to unwrap received message", zap.Error(err))
```

Shouldn't we just return the error? If the message was wrapped, there is no way to process it further anyway no?

---

The error should be returned after the SDS wrap is enable by default, I will disable SDS wrap in the PR after tests, and enable it in newer release as it introduce broken changes.